### PR TITLE
feat(samradsgatt): Kam 806 - add props to case details page and refactor

### PR DESCRIPTION
**Pull request by helgamargreto**
# Add props to case details page and refactor

https://veflausnir.atlassian.net/browse/KAM-806
https://veflausnir.atlassian.net/browse/KAM-844

## What

I added props to necessary components on the case details page.
I made components standalone and make the code more readable  by removing the left, middle and right side components.

## Why

Props missing and too complicated code.

## Screenshots / Gifs

image

https://www.figma.com/file/gFSUsFCIMYWRGa94XsmhpT/DEV-Samr%C3%A1%C3%B0sg%C3%A1tt?node-id=246%3A8033&t=HPNmkoygoUxHVOlR-0

## How to test
Checkout branch feature/samradsgatt/KAM-806-props-add-props-to-case-details-page
Run the code locally
Navigate to http://localhost:4200/samradsgatt/mal/id
Compare desing and the page.
Look at the code and make sure it makes sense :)

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: apps/samradsgatt/types/viewModels.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  processEnds: string
+  announcementText: string
+  summaryDate: string
+  summaryText: string
+  adviceCount: number
+  created: string
+  changed: string
+  oldInstitutionName: string
+}
+
```
Comment (diljath) [inline apps/samradsgatt/types/viewModels.ts:22]: Fínt að hafa þetta núna en mun verða óþarfi í framtíðinni. Allar týpur etc mun verða generate-aðar útfrá api-inum

File: apps/samradsgatt/pages/mal/[slug].tsx [hunk 2]
```diff
@@ -1,31 +1,161 @@
...
+  Divider,
+  CategoryCard,
+} from '@island.is/island-ui/core'
+import { Case, Advice } from '../../types/viewModels'
+import CaseTimeline from '../../../../apps/samradsgatt/components/CaseTimeline/CaseTimeline'
+import SubscriptionBox from '../../../../apps/samradsgatt/components/SubscriptionBox/SubscriptionBox'
+import { useLocation } from 'react-use'
+import CaseOverview from '../../components/CaseOverview/CaseOverview'
+import ReviewCard from '../../components/ReviewCard/ReviewCard'
+import WriteReviewCard from '../../components/WriteReviewCard/WriteReviewCard'
```
Comment (diljath) [inline apps/samradsgatt/pages/mal/[slug].tsx:18]: Það er index skjal í components folderinum sem þú gætir bætt við exportum í svo að þú gætir sótt þetta allt í einni línu. Þá yrði þetta 
import {CaseOverview, WriteReviewCard, ... etc } from '../../components'

File: apps/samradsgatt/components/ReviewCard/ReviewCard.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import { Box, Text } from '@island.is/island-ui/core'
+import { format } from 'date-fns'
```
Comment (diljath) [inline apps/samradsgatt/components/ReviewCard/ReviewCard.tsx:2]: lintið kvartar yfir þessu. Pæling að athuga hvort island-is sé með útfært einhverskonar format?

**Review by diljath — APPROVED** [review]

File: apps/samradsgatt/pages/mal/[slug].tsx (hunk 2 above)
Comment (helgamargreto) [reply apps/samradsgatt/pages/mal/[slug].tsx]: Snilld, buin ad breyta.

File: apps/samradsgatt/types/viewModels.ts (hunk 1 above)
Comment (helgamargreto) [reply apps/samradsgatt/types/viewModels.ts:22]: Akkurat, vildi bara getad profad :)

**Pull request merged**